File: 2. Loop Problems/Loop Problems in c/p23.c
// Write a C program to convert a binary number into decimal.

#include<stdio.h>
#include<math.h>


int main() {
    int n,decimal = 0,remainder,power=0;
    printf("Enter Valid Binary Number: ");
    scanf("%d",&n);

    while ( n!= 0) {
        remainder = n % 10;
        decimal += remainder*pow(2,power);
        n /= 10;
        power++;
    }

    printf("Decimal value: %d",decimal);

}

File: 2. Loop Problems/Loop problems in python/p23.py
# Write a python program to convert a binary number into decimal.


n = int(input("Enter Valid Binary Number: "))



decimal = 0
remainder = 0
position = 0

while n != 0:
    remainder = n % 10
    decimal += remainder*(2**position)
    n = n // 10
    position+=1
    
    

print(f"decimal: {decimal}")